mlflow/tracking/fluent.py

```diff
@@ -1328,9 +1342,26 @@ def _get_experiment_id():
         _active_experiment_id
         or _get_experiment_id_from_env()
         or (is_in_databricks_notebook() and get_notebook_id())
+        or (is_in_databricks_job() and get_job_type_info() == "NORMAL" and _create_job_experiment())
     ) or deprecated_default_exp_id
 
 
+def _create_job_experiment() -> str:
+    job_id = get_job_id()
+    tags = {}
+    tags[MLFLOW_DATABRICKS_JOB_TYPE_INFO] = get_job_type_info()
+    tags[MLFLOW_EXPERIMENT_SOURCE_TYPE] = SourceType.to_string(SourceType.JOB)
+    tags[MLFLOW_EXPERIMENT_SOURCE_ID] = job_id
+
+    experiment_id = create_experiment(get_experiment_name_from_job_id(job_id), None, tags)
+    _logger.info(
+        "Job experiment with experiment_id '%s' created",
+        experiment_id,
+    )
```

Comment:
Users will now see this every time they run MLflow from a job, even if the experiment already existed and nothing new was created. Shall we change this to a `debug()` log?